### DISCUSSION TITLE
Force re-render dom when variable changes

### DISCUSSION THREAD

**Discussion by AbdelElrafa** [alpinejs/alpine#1894] [7. Bugs (archived)]
Hi there,

I'm having a hard time re-rendering the dom when a getter is updated. I'm using a query parameter to track the value and push state to allow the back button to go back. When the back button is clicked the value is not updated in the dom however it is updated programmatically. Is there any way to force the dom to refresh?

Here is a sandbox with some code to test: https://codesandbox.io/s/jolly-ritchie-roqle

```PHP
get counter() {
    const params = new URLSearchParams(location.search);

    return Number(params.get("count") || 0);
},
set counter(value) {
    history.pushState(null, document.title, "?count=" + value);
},

**Comment by KevinBatdorf** [comment]
I think you need to watch for changes. Maybe with this? 

https://developer.mozilla.org/en-US/docs/Web/API/Window/popstate_event

You might also need to extract the getter/setter into methods and update counter as a component property directly for the dom to update

**Reply by KevinBatdorf** [reply]
Checked the code demo again. Did you already have the popstate listener there? My bad. However, you need to update a property directly instead of using a setter. Those won't trigger the proxy trap.
